### PR TITLE
Allow specifying a default value to optional()

### LINKED ISSUE #65: Allow providing a default to optional()
It would be convenient to have `optional(default_value)` which would return `default_value` instead of `None`.
This can be seen as complementary to `result(default_value)` that always return a fixed value.

### PR DESCRIPTION
**Pull request by unode**
Fixes #65

### CODE REVIEW

**Review by spookylukey — CHANGES_REQUESTED** [review]
This looks great thanks, just a small change needed in the docs.

File: docs/ref/methods_and_combinators.rst [hunk 1]
```diff
@@ -141,6 +141,8 @@ can be used and manipulated as below.
          'A'
          >>> string('A').optional().parse('')
          None
+         >>> string('A').optional('Oops').parse('')
+         'Oops'
```
[the remark this answers is not in the document]
Comment (unode) [reply docs/ref/methods_and_combinators.rst:145]: Ah! I missed those. Will update the PR.

[the remark this answers is not in the document]
Comment (unode) [reply docs/ref/methods_and_combinators.rst:146]: Should be done now

File: docs/ref/methods_and_combinators.rst [hunk 2]
```diff
@@ -130,17 +130,20 @@ can be used and manipulated as below.
 
    .. versionadded:: 2.0
 
-   .. method:: optional()
+   .. method:: optional(default=None)
```
Comment (unode) [inline docs/ref/methods_and_combinators.rst:133]: Sphinx syntax here fails me, do I need to use `[default=None]` to indicate that it's an optional argument?

Comment (unode) [reply docs/ref/methods_and_combinators.rst]: I can't tell from the docs so including the `[ ]` for clarity.

Comment (spookylukey) [reply docs/ref/methods_and_combinators.rst]: You actually write it just the way it appears in the sig (apart from `self` parameter). But I'll fix it up, thanks.

**Comment by spookylukey** [comment]
Thanks!

**Pull request merged**